cmd/soci-snapshotter-grpc/main.go

```diff
@@ -139,29 +142,31 @@ func main() {
 		credsFuncs = append(credsFuncs, kubeconfig.NewKubeconfigKeychain(ctx, opts...))
 	}
 	if cfg.CRIKeychainConfig.EnableKeychain {
-		// connects to the backend CRI service (defaults to containerd socket)
-		connectCRI := func() (runtime_alpha.ImageServiceClient, error) {
-			// TODO: make gRPC options configurable from config.toml
-			backoffConfig := backoff.DefaultConfig
-			backoffConfig.MaxDelay = 3 * time.Second
-			connParams := grpc.ConnectParams{
-				Backoff: backoffConfig,
-			}
-			gopts := []grpc.DialOption{
-				grpc.WithTransportCredentials(insecure.NewCredentials()),
-				grpc.WithConnectParams(connParams),
-				grpc.WithContextDialer(dialer.ContextDialer),
-				grpc.WithDefaultCallOptions(grpc.MaxCallRecvMsgSize(defaults.DefaultMaxRecvMsgSize)),
-				grpc.WithDefaultCallOptions(grpc.MaxCallSendMsgSize(defaults.DefaultMaxSendMsgSize)),
+
+		connectV1AlphaCRI := func() (runtime_alpha.ImageServiceClient, error) {
+			criConn, err := getCriConn(cfg.CRIKeychainConfig.ImageServicePath)
+			if err != nil {
+				return nil, err
 			}
-			conn, err := grpc.Dial(dialer.DialAddress(cfg.CRIKeychainConfig.ImageServicePath), gopts...)
+			return runtime_alpha.NewImageServiceClient(criConn), nil
+		}
+
+		connectV1CRI := func() (runtime.ImageServiceClient, error) {
+			criConn, err := getCriConn(cfg.CRIKeychainConfig.ImageServicePath)
 			if err != nil {
 				return nil, err
 			}
-			return runtime_alpha.NewImageServiceClient(conn), nil
+			return runtime.NewImageServiceClient(criConn), nil
 		}
-		f, criServer := cri.NewCRIKeychain(ctx, connectCRI)
-		runtime_alpha.RegisterImageServiceServer(rpc, criServer)
+
+		// register v1alpha2 CRI server with the gRPC server
```

Comment:
Containerd has already deprecated  CRI v1alpha2 since containerd 1.7 https://github.com/containerd/containerd/pull/8276. Do we need to still support v1alpha2 or just replace with v1?

Reply:
K8s >=1.26 dropped support for v1alpha2 https://kubernetes.io/blog/2022/12/09/kubernetes-v1-26-release/#cri-v1alpha2-removed. EKS still supports <1.26 until May 2024 https://docs.aws.amazon.com/eks/latest/userguide/kubernetes-versions.html#kubernetes-release-calendar, so we will have to support at least until then.